### PR TITLE
{tools}[foss/2021a] Horovod v0.25.0 w/ Python 3.9.5 + CUDA 11.3.1 + PyTorch 1.10.0

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3900.accelgor.os - Linux RHEL 8.4, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 510.73.08, Python 3.6.8
See https://gist.github.com/10ce8f030bcfe1400c9b1740fa68c16b for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+easyblock = 'PythonBundle'
+
+name = 'Horovod'
+version = '0.25.0'
+local_pt_version = '1.10.0'
+local_cuda_suffix = '-CUDA-%(cudaver)s'
+versionsuffix = local_cuda_suffix + '-PyTorch-%s' % local_pt_version
+
+homepage = 'https://github.com/uber/horovod'
+description = "Horovod is a distributed training framework for TensorFlow."
```
Comment (akesandgren) [inline easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb:10]: Perhaps the description should be updated to mention PyTorch and mxnet.
Like
```
description = """Horovod is a distributed training framework for TensorFlow, PyTorch and MXnet.
This build only have PyTorch enabled."""
```

**Review by jfgrimm — COMMENTED** [review]
As decided in issue #16330, we have deprecated the use of `True` to signify a system-toolchain dependency (#16384), in favour of the more intuitive `SYSTEM` template constant. Due to the change in the test suite, please run `eb --sync-pr-with-develop 16135` and update the PR to use `SYSTEM` instead.

File: easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+toolchain = {'name': 'foss', 'version': '2021a'}
+
+builddependencies = [
+    ('CMake', '3.20.1'),
+    ('flatbuffers', '2.0.0'),
+]
+dependencies = [
+    ('Python', '3.9.5'),
+    ('PyYAML', '5.4.1'),
+    ('CUDA', '11.3.1', '', True),
```
Comment (branfosj) [inline easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb:22]:
```suggestion
    ('CUDA', '11.3.1', '', SYSTEM),
```

File: easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+name = 'Horovod'
+version = '0.25.0'
+local_pt_version = '1.10.0'
+local_cuda_suffix = '-CUDA-%(cudaver)s'
+versionsuffix = local_cuda_suffix + '-PyTorch-%s' % local_pt_version
+
+homepage = 'https://github.com/uber/horovod'
+description = """Horovod is a distributed training framework for TensorFlow, PyTorch and MXnet.
+This build only have PyTorch enabled."""
```
Comment (branfosj) [inline easybuild/easyconfigs/h/Horovod/Horovod-0.25.0-foss-2021a-CUDA-11.3.1-PyTorch-1.10.0.eb:11]:
```suggestion
This build only has PyTorch enabled."""
```

**Review by branfosj — CHANGES_REQUESTED** [review]

**Review by branfosj — APPROVED** [review]

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0203u29a.bear.cluster - Linux RHEL 8.6, x86_64, Intel(R) Xeon(R) Platinum 8360Y CPU @ 2.40GHz (icelake), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 520.61.05, Python 3.6.8
See https://gist.github.com/branfosj/ad2603847a816c3b780307cc6abe7b2b for a full test report.

**Comment by branfosj** [comment]
Going in, thanks @boegel!

**Pull request merged**